lib/services/local/notifications.go

```diff
@@ -0,0 +1,298 @@
+	// this field will have been populated.
+	if notification.Spec.Created != nil {
+		return nil, trace.Wrap(trace.BadParameter("notification created time should be empty"))
+	}
+	notification.Spec.Created = timestamppb.New(time.Now())
+
+	// Append username prefix.
+	serviceWithPrefix, err := s.userNotificationService.WithPrefix(username)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	created, err := serviceWithPrefix.CreateResource(ctx, notification)
+	return created, trace.Wrap(err)
+}
+
+// DeleteUserNotification deletes a user-specific notification.
+func (s *NotificationsService) DeleteUserNotification(ctx context.Context, username string, notificationId string) error {
+	// Append username prefix.
+	serviceWithPrefix, err := s.userNotificationService.WithPrefix(username)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = serviceWithPrefix.DeleteResource(ctx, notificationId)
+	return trace.Wrap(err)
+}
+
+// DeleteAllUserNotificationsForUser deletes all of a user's user-specific notifications.
+func (s *NotificationsService) DeleteAllUserNotificationsForUser(ctx context.Context, username string) error {
+	// Append username prefix.
+	serviceWithPrefix, err := s.userNotificationService.WithPrefix(username)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = serviceWithPrefix.DeleteAllResources(ctx)
+	return trace.Wrap(err)
+}
+
+// CreateGlobalNotification creates a global notification.
+func (s *NotificationsService) CreateGlobalNotification(ctx context.Context, globalNotification *notificationsv1.GlobalNotification) (*notificationsv1.GlobalNotification, error) {
+	if err := services.ValidateGlobalNotification(globalNotification); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	globalNotification.Kind = types.KindGlobalNotification
+	globalNotification.Version = types.V1
+
+	if globalNotification.Metadata == nil {
+		globalNotification.Metadata = &headerv1.Metadata{}
+	}
+
+	// We set this to the UUID because the service adapter uses `getName()` to determine the backend key to use when storing the notification.
+	globalNotification.Metadata.Name = globalNotification.Spec.Notification.Spec.Id
+
+	if err := CheckAndSetExpiry(&globalNotification.Spec.Notification.Metadata.Expires); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if globalNotification.Spec.Notification.Spec.Created != nil {
+		return nil, trace.Wrap(trace.BadParameter("notification created time should be empty"))
+	}
+	globalNotification.Spec.Notification.Spec.Created = timestamppb.New(time.Now())
+
+	created, err := s.globalNotificationService.CreateResource(ctx, globalNotification)
+	return created, trace.Wrap(err)
+}
+
+// DeleteGlobalNotification deletes a global notification.
+func (s *NotificationsService) DeleteGlobalNotification(ctx context.Context, notificationId string) error {
+	err := s.globalNotificationService.DeleteResource(ctx, notificationId)
```

Comment:
Hmm, we would want to delete the notification states for this notification, but I don't think there is a performant way to do it here. We would need to list all notification states for all users, filter them for ones that are for this notification, and then loop through them and delete each one. I think this might be better if this was done during a periodic cleanup, what do you think?